staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,297 @@
+	// request flow identifier’s string pair is hashed and the hash value is used to shuffle the queue indices and deal
+	// a hand of the size specified here. If empty, the hand size will the be set to 1.
+	// +optional
+	HandSize int32 `json:"handSize,omitempty" protobuf:"varint,4,opt,name=handSize"`
+	// `queueLengthLimit` is a length limit applied to each queue belongs to the priority.  The value must be positive
+	// for a non-exempt priority level.
+	// +optional
+	QueueLengthLimit int32 `json:"queueLengthLimit,omitempty" protobuf:"varint,5,opt,name=queueLengthLimit"`
+}
+
+// PriorityLevelConfigurationConditionType is a valid value for PriorityLevelConfigurationStatusCondition.Type
+type PriorityLevelConfigurationConditionType string
+
+// PriorityLevelConfigurationStatus represents the current state of a "request-priority".
+type PriorityLevelConfigurationStatus struct {
+	// `conditions` is the current state of "request-priority".
+	Conditions []PriorityLevelConfigurationCondition `json:"conditions,omitempty" protobuf:"bytes,1,rep,name=conditions"`
+}
+
+// PriorityLevelConfigurationCondition defines the condition of priority level.
+type PriorityLevelConfigurationCondition struct {
+	// `type` is the type of the condition.
+	Type PriorityLevelConfigurationConditionType `json:"type,omitempty" protobuf:"bytes,1,opt,name=type"`
+	// `status` is the status of the condition.
+	// Can be True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status,omitempty" protobuf:"bytes,2,opt,name=status"`
+	// `lastTransitionTime` is the last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,3,opt,name=lastTransitionTime"`
+	// `reason` is a unique, one-word, CamelCase reason for the condition's last transition.
+	Reason string `json:"reason,omitempty" protobuf:"bytes,4,opt,name=reason"`
+	// `message` is a human-readable message indicating details about last transition.
+	Message string `json:"message,omitempty" protobuf:"bytes,5,opt,name=message"`
+}
+
+// Subject matches a set of users.
+// Syntactically, Subject is a general API object reference.
+// Authorization produces a username and a set of groups, and we imagine special kinds of non-namespaced objects,
+// User and Group in API group "flowcontrol.apiserver.k8s.io", to represent such a username or group.
+// The only kind of true object reference that currently will match any users is ServiceAccount.
+type Subject struct {
+	// `kind` of object being referenced. Values defined by this API group are "User", "Group", and "ServiceAccount".
+	// If the kind value is not recognized, the flow-control layer in api-server should report an error.
+	Kind string `json:"kind" protobuf:"bytes,1,opt,name=kind"`
+	// `apiGroup` holds the API group of the referenced subject.
+	// Defaults to "" for ServiceAccount subjects.
+	// Defaults to "flowcontrol.apiserver.k8s.io" for User and Group subjects.
```

Comment:
We should maybe use a value that's more obviously TBD-- we would not actually store users or groups in that api group.

Reply:
This is exactly analogous to what happens in rbac.

Reply:
:/ OK

Reply:
I tried to say pretty clearly that there are special Kinds used here that do not correspond to "real" objects.  If you can suggest a clearer or more emphatic way to communicate the idea...

Reply:
yes, we're trying to keep in consistent w/ rbac here

Reply:
BTW, as I have noted elsewhere, I think the corresponding type comment in rbac is even less clear than what we have here regarding this matter.